modules/gapi/include/opencv2/gapi/rmat.hpp

```diff
@@ -0,0 +1,127 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#ifndef OPENCV_GAPI_RMAT_HPP
+#define OPENCV_GAPI_RMAT_HPP
+
+#include <opencv2/gapi/gmat.hpp>
+
+namespace cv {
+
+// A lightweight wrapper on image data:
+// - Doesn't own the memory;
+// - Doesn't implement copy semantics (it's assumed that a view is created each time
+// wrapped data is being accessed);
+// - Has an optional callback which is called when the view is destroyed.
+class MatView
```

Comment:
> MatView

`cv::MatView` doesn't look as a right name.
Please apply "G" prefix.

Or move to `gapi` namespace.

/cc @dmatveev

Reply:
@alalek  Why not?

Our "G"-prefixed types are graph types which serve for graph building purposes, they never hold the real data.

Would `cv::RMat::View` work instead? I believe `cv::RMat` itself is quite useful to be part of `core`, not just `gapi` module. Maybe consider this for 5.0?

Reply:
> `cv::RMat::View`

Looks good to me.


> to be part of core

Problem with `GMatDesc` dependency and other gapi structures should be resolved first (`core` modules should not depend on `gapi`).

---

Actually I thought that `RMat`/`GMat`/etc are in `cv::gapi` namespace.

Reply:
> Actually I thought that RMat/GMat/etc are in cv::gapi namespace.

This would make the code too verbose. We decided to keep data structures and classes (which we have <10) under `cv::` and 100s of kernels/functions under `cv::gapi`.

Reply:
> Problem with GMatDesc dependency and other gapi structures should be resolved first (core modules should not depend on gapi).

Btw, a structure _describing_ Mat is worth having at the core OpenCV too. :)

Reply:
Moved: MatView -> RMat::View